src/ImageSharp/Formats/Tiff/Utils/TiffUtils.cs

```diff
@@ -126,6 +151,27 @@ public static TPixel ColorScaleTo32Bit<TPixel>(ulong intensity, TPixel color)
             return color;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void UnPremultiplyRow<TPixel>(Span<Vector4> vectors, Span<TPixel> pixelRow, TPixel color)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            for (int x = 0; x < vectors.Length; x++)
+            {
+                Vector4 vec = vectors[x];
+                pixelRow[x] = UnPremultiply(vec, color);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static TPixel UnPremultiply<TPixel>(Vector4 vec, TPixel color)
```

Comment:
Do you call this method when not using  collection? We have `Numerics.UnPremultiply(ref Vector4 vector)`